exaudfclient/docs/script_options_design.md

```diff
@@ -0,0 +1,466 @@
+        ctx.emit(\"Success!\");
+    }
+}
+```
+
+_other_script_A_:
+```
+%import other_script_B;
+class OtherClassA {
+    static void doSomething() {}
+}
+```
+
+_other_script_B_:
+```
+class OtherClassB {
+    static void doSomething() {}
+}
+```
+
+_other_script_C_:
+```
+%import other_script_A
+class OtherClassC {
+    static void doSomething() {}
+}
+```
+
+The following diagram shows how the scripts are collected in the recursive algorithm:
+
+![V2ImportScriptFlow](diagrams/V2ImportScriptFlow.drawio.png)
+
+
+
+## Runtime
+
+### Parser Implementation V1
+
+The legacy parser (V1) parser searches for one specific script option. The parser starts from beginning of the script code. If found, the parser immediately removes the script option from the script code and returns the option value. It validates the 
+
+### Parser Implementation V2
+
+The parser provides an interface to parse the UDF script for all options at once. All found script options need to be collected in an associative container. Internally. the parses uses [ctpg](https://github.com/peter-winter/ctpg) to parse the UDF script code line-by-line.
+
+## Cross-cutting Concerns
+
+
+## Design Decisions for V2
+
+### Parser Implementation
+`dsn~parser-implementation~1`
+
+Implement the parser using [ctpg](https://github.com/peter-winter/ctpg), an open-source parser library. This library will be used to define Lexer and Parser Rules in C++ code, ensuring no additional runtime dependencies exist.
+
+
+Covers:
+- `req~general-script-options-parsing~1`
+- `req~existing-parser-library-license~1`
+
+Tags: V2
+
+
+### Lexer and Parser Rules Option
+`dsn~lexer-parser-rules~1`
+
+Lexer and Parser rules to recognize `%optionKey`, `optionValue`, with whitespace characters as separator. The Parser rules will define the grammar to correctly identify Script Options, manage multiple options with the same key, and handle duplicates.
+
+
+Covers:
+- `req~general-script-options-parsing~1`
+- `req~white-spaces~1`
+- `req~leading-white-spaces-script-options-parsing~1`
+
+Depends:
+- `dsn~parser-implementation~1`
+
+Tags: V2
+
+### Lexer and Parser Rules Not an option
+`dsn~lexer-parser-rules-not-an-option~1`
+
+Lexer and Parser rules to recognize anything what is not an option.
+
+
+Covers:
+- `req~ignore-none-script-options~1`
+
+Depends:
+- `dsn~parser-implementation~1`
+
+Tags: V2
+
+### Run Parser line-by-line
+`dsn~run-parser-line-by-line~1`
+
+The parser must be executed line-by-line, because script options can be placed at any location in the UDF. Every found map of options must be added to the resulting map.
+
+
+Covers:
+- `req~multiple-lines-script-options-parsing~1`
+
+Depends:
+- `dsn~parser-implementation~1`
+
+Tags: V2
+
+### Ignore lines without script options
+`dsn~ignore-lines-without-script-options~1`
+
+In order to avoid lower performance compared to the old implementation, the parser must run only on lines which contain a `%` character.
```

Comment:
```suggestion
In order to avoid lower performance compared to the old implementation, the parser must run only on lines which contain a `%` character after only whitespaces 
```